src/spetlr/power_bi/SparkPandasDataFrame.py

```diff
@@ -0,0 +1,314 @@
+from datetime import datetime
+from typing import Callable, Dict, List, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from dateutil.parser import parse
+from pandas.core.generic import NDFrameT
+from pyspark.sql import DataFrame
+from pytz import timezone, utc
+
+from spetlr.exceptions import SpetlrException
+from spetlr.spark import Spark
+
+
+class SparkPandasDataFrame:
+    # converts data types from Spark to Pandas and vice versa
+    SPARK_TO_PANDAS_TYPES = {
+        "long": "Int64",
+        "int": "Int64",
+        "timestamp": "datetime64[ns]",
+        "boolean": "bool",
+        "string": "object",
+        "default": "object",
+    }
+
+    def __init__(
+        self,
+        json: Union[Dict, List],
+        schema: List[Tuple[Union[str, Callable[[pd.DataFrame], NDFrameT]], str, str]],
+        *,
+        indexing_columns: Union[int, List[int], str, List[str], None] = None,
+        sorting_columns: Union[int, List[int], str, List[str], None] = None,
+        ascending: bool = True,
+        local_timezone_name: str = None,
+    ):
+        """
+        Provides a class for loading a JSON into Pandas, and for converting
+        between Pandas and Spark.
+
+        :param dict json: Data source, which is a dictionary representing a JSON.
+        :param dict[tuple[str/callable, str, str]: schema definition where the first
+            value in the tuple is the original column name in the JSON, the second
+            is the desired column name in the output, and the third is the Spark type.
+            The first value can also be a lambda expression taking a Pandas
+            data frame as input and calculating the column value from
+            other available columns, e.g. lambda df: df["endTime"] - df["startTime"]
+        :param int/str indexing_columns: indexes (within the schema) or names
+            of the indexing columns.
+        :param int/str sorting_columns: indexes (within the schema) or names
+            of the sorting columns.
+        :param str local_timezone_name: The timezone to use when parsing
+            timestamp columns. The default timezone is UTC.
+            If the timezone is UTC, all timestamp columns will have a suffix "Utc".
+            Otherwise they will have a suffix "Local".
+        """
+
+        self.local_timezone_name = (
```

Comment:
self.local_timezone_name = local_timezone_name or "UTC"

Reply:
I changed it. :)